apps/gateway/src/utils/converters.ts

```diff
@@ -105,12 +105,30 @@ export function toModelMessages(messages: OpenAICompatibleMessage[]) {
         if (message.tool_calls) {
           modelMessages.push({
             role: "assistant",
-            content: message.tool_calls.map((toolCall) => ({
-              type: "tool-call",
-              toolCallId: toolCall.id,
-              toolName: toolCall.function.name,
-              input: JSON.parse(toolCall.function.arguments),
-            })),
+            content: message.tool_calls.map((toolCall) => {
+              const {
+                // eslint-disable-next-line @typescript-eslint/no-unused-vars
+                type,
+                id,
+                function: toolFunction,
+                ...metadata
+              } = toolCall;
+
+              const content: any = {
+                type: "tool-call",
+                toolCallId: id,
+                toolName: toolFunction.name,
+                input: JSON.parse(toolFunction.arguments),
+              };
+
+              if (Object.keys(metadata).length > 0) {
```

Comment:
Why are you moving the metadata from the content part into the providerOptions? Isn't the metadata part-specific and the provider options are call-level?

Reply:
AI SDK does accept providerOptions at content-part level, which is not being mentioned in the documentation. 

FYR: https://github.com/vercel/ai/blob/main/packages/ai/src/prompt/content-part.ts#L82